### PR TITLE
Additional class documentation

### PR DESCRIPTION
**Pull request by LengthenedGradient**
Following up from the previous documentation update, this one should document the remainder of the base class files.

### CODE REVIEW

File: lua/acf/core/classes/grouped.lua [hunk 1]
```diff
@@ -80,6 +94,9 @@ function Classes.GetGroup(Namespace, ID)
 	end
 end
 
+--- Indexes the groups stored in Entries into a new Namespace, with helper functions
+--- @param Namespace table The namespace to store the groups under
```
Comment (TwistedTail) [inline lua/acf/core/classes/grouped.lua:98]: The namespace in this case will be the table that will receive the new functions.

File: lua/acf/core/classes/grouped.lua [hunk 2]
```diff
@@ -63,6 +73,10 @@ function Classes.AddGroupItem(ID, GroupID, Destiny, Data)
 	return Class
 end
 
+--- Gets a group or group item given its ID and the namespace it's stored in
+--- @param Namespace string The table to store the group in
```
Comment (TwistedTail) [inline lua/acf/core/classes/grouped.lua:77]: On this one, the namespace will be used to look for a group

File: lua/acf/core/classes/grouped.lua [hunk 3]
```diff
@@ -4,7 +4,11 @@ local isstring = isstring
 local istable  = istable
 local Classes  = ACF.Classes
 
-
+--- Registers a group
+--- @param ID string The ID of the group
+--- @param Destiny table The table to store the group in
+--- @param Data table The data of the group
+--- @return table Group The created group
```
Comment (TwistedTail) [inline lua/acf/core/classes/grouped.lua:11]: This value can be nil under certain cases

File: lua/acf/core/classes/grouped.lua [hunk 4]
```diff
@@ -32,6 +36,12 @@ function Classes.AddGroup(ID, Destiny, Data)
 	return Group
 end
 
+--- Registers a group item under a group
+--- @param ID string The ID of the item to add to the group
+--- @param GroupID string The ID of the group previously created using Class.AddGroup
+--- @param Destiny string The table to store the group item in (should be the same as the group)
+--- @param Data table The data of the group item
+--- @return table Class The created group item
```
Comment (TwistedTail) [inline lua/acf/core/classes/grouped.lua:44]: This value can be nil under certain conditions

File: lua/acf/core/classes/object.lua [hunk 5]
```diff
@@ -74,7 +74,7 @@ end
 --- @param ID string The ID of the new sub class to add
 --- @param Base string The ID of the base class the sub class will inherit from
 --- @param Destiny table A table that the new object will be indexed into, with the ID as key
---- @return table # The created class
+--- @return table # The created object
```
Comment (TwistedTail) [inline lua/acf/core/classes/object.lua:77]: This value can be nil under certain conditions

File: lua/acf/core/classes/simple.lua [hunk 6]
```diff
@@ -4,7 +4,12 @@ local isstring = isstring
 local istable  = istable
 local Classes  = ACF.Classes
 
-
+--- Registers a simple class 
+--- Similar to Classes.AddObject
+--- @param ID string The ID of the simple class to add
+--- @param Destiny table The table to store the simple class in
+--- @param Data table The data of the simple class
+--- @return table Class The created simple class
```
Comment (TwistedTail) [inline lua/acf/core/classes/simple.lua:12]: This value can be nil under certain conditions

File: lua/acf/core/classes/simple.lua [hunk 7]
```diff
@@ -29,6 +34,9 @@ function Classes.AddSimple(ID, Destiny, Data)
 	return Class
 end
 
+--- Indexes the simple classes stored in Entries into a new Namespace, with helper functions
+--- @param Namespace table The namespace to store the simple classes under
```
Comment (TwistedTail) [inline lua/acf/core/classes/simple.lua:38]: Same as the grouped ones, the namespace given to this function will receive a new set of functions

**Review by TwistedTail — CHANGES_REQUESTED** [review]
Seems fine, just a few missing things and a couple misleading comments in relation to Add*Functions methods.

**Review by TwistedTail — APPROVED** [review]
Looks fine now

**Comment by LengthenedGradient** [comment]
As of the namespace documentation update, In various places I have things like (objects/simple classes/groups) written. It makes no sense to call them classes because an object is not a class. What else could I call it?

**Comment by TwistedTail** [comment]
> In various places I have things like (objects/simple classes/groups) written. It makes no sense to call them classes because an object is not a class. What else could I call it?

Personally I would just refer to them as "entries", but I suppose someone else could come up with a better term.

**Comment by LengthenedGradient** [comment]
> > In various places I have things like (objects/simple classes/groups) written. It makes no sense to call them classes because an object is not a class. What else could I call it?
> 
> Personally I would just refer to them as "entries", but I suppose someone else could come up with a better term.

I just went with entries, that should work.

**Comment by TwistedTail** [comment]
Anything else you might want to add/change? It seems alright.

**Comment by LengthenedGradient** [comment]
> Anything else you might want to add/change? It seems alright.

I think this is ready to be merged. I don't think I'll need to add/change anything else.

**Review by thecraftianman — APPROVED** [review]

**Pull request merged**